File: tests/human/x/ts/left_join.ts
const customers = [
  { id: 1, name: "Alice" },
  { id: 2, name: "Bob" },
];
const orders = [
  { id: 100, customerId: 1, total: 250 },
  { id: 101, customerId: 3, total: 80 },
];
interface Entry {
  orderId: number;
  customer: { id: number; name: string } | undefined;
  total: number;
}
const result: Entry[] = orders.map((o) => ({
  orderId: o.id,
  customer: customers.find((c) => c.id === o.customerId),
  total: o.total,
}));
console.log("--- Left Join ---");
for (const entry of result) {
  console.log("Order", entry.orderId, "customer", entry.customer, "total", entry.total);
}

File: tests/human/x/python/left_join.py
customers = [
    {"id": 1, "name": "Alice"},
    {"id": 2, "name": "Bob"},
]
orders = [
    {"id": 100, "customerId": 1, "total": 250},
    {"id": 101, "customerId": 3, "total": 80},
]

result = []
for o in orders:
    c = next((c for c in customers if c["id"] == o["customerId"]), None)
    result.append({"orderId": o["id"], "customer": c, "total": o["total"]})

print("--- Left Join ---")
for entry in result:
    print("Order", entry["orderId"], "customer", entry["customer"], "total", entry["total"])
